test/integration/userStatus.test.js

```diff
@@ -554,6 +556,135 @@ describe("UserStatus", function () {
     });
   });
 
+  describe("PATCH /users/status/self", function () {
+    let userId;
+    let userJwt;
+
+    before(async function () {
+      userId = await addUser(userData[8]);
+      userJwt = authService.generateAuthToken({ userId });
+    });
+
+    afterEach(async function () {
+      await firestore.collection("tasks").doc("user1AssignedTask").delete();
+      await firestore.collection("usersStatus").doc("user1AssignedStatus").delete();
+    });
+
+    after(async function () {
+      await firestore.collection("users").doc(userId).delete();
+    });
+
+    it("Should Change the status to ACTIVE if user has task assigned.", async function () {
+      const now = new Date();
+      const nowTimeStamp = new Date().setUTCHours(0, 0, 0, 0);
+      const fiveDaysFromNowTimeStamp = new Date(now.setUTCHours(0, 0, 0, 0) + 5 * 24 * 60 * 60 * 1000);
+
+      await firestore.collection("tasks").doc("user1AssignedTask").set({
+        assignee: userId,
+        status: "ASSIGNED",
+      });
+      await firestore
+        .collection("usersStatus")
+        .doc("user1AssignedStatus")
+        .set({
+          userId: userId,
+          currentStatus: {
+            message: "",
+            from: nowTimeStamp,
+            until: fiveDaysFromNowTimeStamp,
+            updatedAt: nowTimeStamp,
+            state: userState.OOO,
+          },
+        });
+      const res = await chai
+        .request(app)
+        .patch(`/users/status/self`)
+        .set("cookie", `${cookieName}=${userJwt}`)
+        .send({ cancelOoo: true });
+      expect(res.body.data.currentStatus.state).to.equal(userState.ACTIVE);
+      expect(res.body.data.currentStatus.from).to.be.gt(nowTimeStamp);
+      expect(res.body.data.currentStatus.until).to.equal("");
+      expect(res.body.data.currentStatus.message).to.equal("");
+    });
+
+    it("Should Change the status to IDLE if user doesn't have a task assigned.", async function () {
+      const now = new Date();
+      const nowTimeStamp = new Date().setUTCHours(0, 0, 0, 0);
+      const fiveDaysFromNowTimeStamp = new Date(now.setUTCHours(0, 0, 0, 0) + 5 * 24 * 60 * 60 * 1000);
```

Comment:
same as above

Reply:
The variables declared in this `describe` block are only needed in two of the `it` blocks. To avoid unnecessary clutter, I have moved the variables to the specific `it` blocks where they are needed. Let me know if you are ok with this